libs/sdk-model/src/insight/index.ts

```diff
@@ -579,6 +587,29 @@ export function insightSetBuckets<T extends IInsightDefinition>(
     } as T;
 }
 
+/**
+ * Creates a new insight with modified bucket items (retrieved by applying the modifications function to each bucketItem in the insight).
+ *
```

Comment:
please expand this doc with something like this:

`Note: the bucket item modification function SHOULD NOT modify bucket item's localId. The localId MAY be used to reference the item from other places in the insight (for example from sorts). Changing the item localId has potential to break the insight: as-is this function does not concern itself with changing the references.`

Reply:
Thanks, I've updated the PR.